src/components/List/AddCard/AddCard.js

```diff
@@ -21,17 +21,25 @@ const AddCard = ({ listId }) => {
   };
 
   const handleAddCard = () => {
-    setOpen(false);
-    setText("");
-    dispatch(addCard({ listId, content: text }));
+    if (text.trim.length !== 0) { 
+      setOpen(false);
+      setText("");
+      dispatch(addCard({ listId, content: text }));
+    }
   };
 
-  const handleKeyPress = (e) => {
-    if (e.code === 'Enter') {
-      handleAddCard();
+  const handleKeyDown = (e) => {
+    switch (e.code) {
+      case 'Enter':
+        handleAddCard();
+        break;
+      case 'Escape':
+        handleClose();
+        break;
     }
```

Comment:
The difference between the onKeyDown and onKeyPress is that the onKeyDown captures keys such as Ctrl, Caps Lock, Escape and Alt, whereas the onKeyPress key will only capture keys that will produce a character on the screen.

Reply:
Ah interesting, I think some places do use keyDown, but there are inconsistencies. Might be worth raising an issue to clear up these inconsistencies if you feel its useful